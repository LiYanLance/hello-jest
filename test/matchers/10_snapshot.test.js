import React from "react";
import renderer from "react-test-renderer";

describe("SnapShot Matchers", () => {
  it("snapshot", () => {
    const SomeComponent = ({ content }) => (
      <div className="some class">{content}</div>
    );

    const tree = renderer
      .create(<SomeComponent content="dummy content" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("inline snapshot", () => {
    const SomeComponent = ({ content }) => (
      <div className="some class">{content}</div>
    );

    const tree = renderer
      .create(<SomeComponent content="dummy content" />)
      .toJSON();

    // Need `prettier` as peer dependency
    expect(tree).toMatchInlineSnapshot(`
      <div
        className="some class"
      >
        dummy content
      </div>
    `);
  });
  
  it("property matcher", () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: "Li Yan",
    };
  
    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      id: expect.any(Number),
    });
  })
  

  describe("customize snapshot", () => {
    
    const { toMatchSnapshot } = require("jest-snapshot")
    
    expect.extend({
      toMatchTrimmedSnapshot(received) {
        delete received.props.className
        return toMatchSnapshot.call(this, received, 'toMatchTrimmedSnapshot')
      }
    })
    
    it("snapshot without classname and spaces", () => {
      const SomeComponent = ({ content }) => (
        <div className="some class">{content}</div>
      );
  
      const tree = renderer
      .create(<SomeComponent content="dummy content" />)
      .toJSON();

      expect(tree).toMatchTrimmedSnapshot()
    })
  })
});
