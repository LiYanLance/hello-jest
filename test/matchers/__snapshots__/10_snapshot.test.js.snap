// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Matchers property matcher 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "Li Yan",
}
`;

exports[`SnapShot Matchers snapshot 1`] = `
<div
  className="some class"
>
  dummy content
</div>
`;
